infra/serverless/main.tf

```diff
@@ -24,3 +29,186 @@ module "lambda_function" {
     Name = "serverless-pricing"
   }
 }
+
+#Lambda IAM Permissions for API Gateway
+resource "aws_lambda_permission" "pricing_agw_perm" {
+  statement_id  = "AllowExecutionFromAPIGateway"
+  action        = "lambda:InvokeFunction"
+  function_name = module.lambda_function.lambda_function_name
+  principal     = "apigateway.amazonaws.com"
+
+  source_arn = "arn:aws:execute-api:${var.my_region}:${var.account_id}:${aws_api_gateway_rest_api.pricing_agw.id}/*/${aws_api_gateway_method.get_method.http_method}${aws_api_gateway_resource.ec2_pricing.path}"
+}
+
+########################################################################################
+# API GATEWAY SETUP
+########################################################################################
+
+#API Gateway
+resource "aws_api_gateway_rest_api" "pricing_agw" {
+  name = "serverless-pricing-api-gateway"
+}
+
+resource "aws_api_gateway_resource" "ec2_pricing" {
+  parent_id   = aws_api_gateway_rest_api.pricing_agw.root_resource_id
+  rest_api_id = aws_api_gateway_rest_api.pricing_agw.id
+  path_part   = "ec2-pricing"
+}
+
+#API Gateway GET Resource
+resource "aws_api_gateway_method" "get_method" {
+  resource_id   = aws_api_gateway_resource.ec2_pricing.id
+  rest_api_id   = aws_api_gateway_rest_api.pricing_agw.id
+  authorization = "NONE"
```

Comment:
In the future you'll change this to CUSTOM or AWS_IAM i'd imagine

Reply:
Yeah, would assume so, but bear in mind this will be called from front end (html/JS), but potentially for unauthenticated journeys, not just authenticated ones